game/src/game/player.rs

```diff
@@ -163,20 +165,20 @@ fn setup_player(q: Query<(&Transform, Entity), Added<PlayerBlueprint>>, mut comm
                 marker: PlayerGent { e_gfx },
                 phys: GentPhysicsBundle {
                     rb: RigidBody::Kinematic,
-                    collider: Collider::cuboid(6.0, 10.0),
+                    collider: Collider::cuboid(4.0, 10.0),
+                    shapecast: ShapeCaster::new(
+                        Collider::cuboid(3.99, 10.0),
+                        Vec2::new(0.0, -2.0),
+                        0.0,
+                        Vec2::NEG_Y.into(),
+                    ),
                 },
+                coyote_time: Default::default(),
             },
             Health {
                 current: 100,
                 max: 100,
             },
-            //get rid of this, move to shapecast with spatialquery
-            ShapeCaster::new(
-                Collider::cuboid(6.0, 10.0),
-                Vec2::new(0.0, -2.0),
-                0.0,
-                Vec2::NEG_Y.into(),
-            ),
```

Comment:
Ended up doing a hybrid approach that works well: a ground based shape cast attached as component for detecting grounded states, and a spatialquery for handling collision logic.